Search v1

caporta opened the pull request:
@SophieDeBenedetto 👀

caporta committed:
instantiate searches_controller + route

caporta committed:
convert HTML form to rails helper

caporta committed:
get searches_controller#new to render searches/new view

caporta committed:
fix funky query param behavior
by slimming down route to just /search. would like to figure out how to include nicely formatted query string in URL

caporta committed:
build out front-end for basic search
use jQuery && SCSS to dynamically show and hide the cohorts select field
&& assure params are pushing to controller as desired.

caporta committed:
add requisite assets for 8b7e9eb

caporta committed:
build out basic controller && view logic for search
add basic styles && seed db with Student record

caporta committed:
improve /search view
by formatting search results as a list && dynamically rendering search type

SophieDeBenedetto commented on app/controllers/searches_controller.rb:
```diff
@@ -0,0 +1,15 @@
+    end
+    render 'new'
+  end
+
```
Create a Search model that has attributes #type, #result, #query, etc. search instance knows its type and can execute student or lab based search accordingly. pass search instance or search result instance in to view.

SophieDeBenedetto commented on app/views/layouts/application.html.erb:
```diff
@@ -26,12 +26,15 @@
+              <% end %>
             </li>
           </ul>
       </div>
```
refactor this navbar form into a partial. maybe navbar also gets its own partial

SophieDeBenedetto commented on app/views/searches/new.html.erb:
```diff
@@ -0,0 +1,17 @@
+<% else %>
+  <p>No <%= @search_type %> matching your query. Please revise your search term.</p>
+<% end %>
+
```
instead, we'll be operating on search or search result instance. So, logic around student/lab results should be refactored into instance method on search class.

SophieDeBenedetto commented on config/routes.rb:
```diff
@@ -3,10 +3,13 @@
+
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
   # Serve websocket cable requests in-process
```
provide url helper by aliasing /search route.

caporta committed:
create url helper by aliasing /search route

caporta committed:
refactor search functionality into standalone model

caporta committed:
add requisite model for commit 770399a

caporta committed:
break navbar and search component into shared partials

caporta committed:
merge master + resolve conflicts

SophieDeBenedetto committed:
Merge branch 'master' of github.com:flatiron-school/flatiron-lab-trackr into search

SophieDeBenedetto committed:
refactor searches controller code into Search instance methods
#execute method contains logic around lab vs. student search, enacts search and sets results

SophieDeBenedetto committed:
refactor search model to fix arg error in call to #lab_search; dependency issue with concurrent-ruby and puffing bill gems. unresolved as of now
